src/pages/audio-rewards-page/store/sagas.ts

```diff
@@ -199,8 +204,13 @@ function* claimChallengeRewardAsync(
       yield put(claimChallengeRewardSucceeded())
     }
   } catch (e) {
-    console.error('Error claiming rewards:', e)
-    yield put(claimChallengeRewardFailed())
+    // Retry once in the case of generic failure, otherwise log error and abort
+    if (retryOnFailure) {
+      yield put(claimChallengeReward({ claim, retryOnFailure: false }))
```

Comment:
This will retry if a user is blocked.

Reply:
Maybe move this logic to default case/UNKNOWN_ERROR case instead?

Reply:
Sounds good, I also removed the default in the switch statement b/c we handle all the cases